crates/oxc_linter/src/rules/eslint/prefer_object_has_own.rs

```diff
@@ -0,0 +1,400 @@
+        "Object.hasOwnProperty.call",
+        "foo.Object.hasOwnProperty.call(obj, prop)",
+        "foo.hasOwnProperty.call(obj, prop)",
+        "foo.call(Object.prototype.hasOwnProperty, Object.prototype.hasOwnProperty.call)",
+        "Object.foo.call(obj, prop)",
+        "Object.hasOwnProperty.foo(obj, prop)",
+        "Object.hasOwnProperty.call.foo(obj, prop)",
+        "Object[hasOwnProperty].call(obj, prop)",
+        "Object.hasOwnProperty[call](obj, prop)",
+        "class C { #hasOwnProperty; foo() { Object.#hasOwnProperty.call(obj, prop) } }",
+        "class C { #call; foo() { Object.hasOwnProperty.#call(obj, prop) } }",
+        "(Object) => Object.hasOwnProperty.call(obj, prop)",
+        "Object.prototype",
+        "Object.prototype(obj, prop)",
+        "Object.prototype.hasOwnProperty",
+        "Object.prototype.hasOwnProperty(obj, prop)",
+        "Object.prototype.hasOwnProperty.call",
+        "foo.Object.prototype.hasOwnProperty.call(obj, prop)",
+        "foo.prototype.hasOwnProperty.call(obj, prop)",
+        "Object.foo.hasOwnProperty.call(obj, prop)",
+        "Object.prototype.foo.call(obj, prop)",
+        "Object.prototype.hasOwnProperty.foo(obj, prop)",
+        "Object.prototype.hasOwnProperty.call.foo(obj, prop)",
+        "Object.prototype.prototype.hasOwnProperty.call(a, b);",
+        "Object.hasOwnProperty.prototype.hasOwnProperty.call(a, b);",
+        "Object.prototype[hasOwnProperty].call(obj, prop)",
+        "Object.prototype.hasOwnProperty[call](obj, prop)",
+        "class C { #hasOwnProperty; foo() { Object.prototype.#hasOwnProperty.call(obj, prop) } }",
+        "class C { #call; foo() { Object.prototype.hasOwnProperty.#call(obj, prop) } }",
+        "Object[prototype].hasOwnProperty.call(obj, prop)",
+        "class C { #prototype; foo() { Object.#prototype.hasOwnProperty.call(obj, prop) } }",
+        "(Object) => Object.prototype.hasOwnProperty.call(obj, prop)",
+        "({})",
+        "({}(obj, prop))",
+        "({}.hasOwnProperty)",
+        "({}.hasOwnProperty(prop))",
+        "({}.hasOwnProperty(obj, prop))",
+        "({}.hasOwnProperty.call)",
+        "({}).prototype.hasOwnProperty.call(a, b);",
+        "({}.foo.call(obj, prop))",
+        "({}.hasOwnProperty.foo(obj, prop))",
+        "({}[hasOwnProperty].call(obj, prop))",
+        "({}.hasOwnProperty[call](obj, prop))",
+        "({}).hasOwnProperty[call](object, property)",
+        "({})[hasOwnProperty].call(object, property)",
+        "class C { #hasOwnProperty; foo() { ({}.#hasOwnProperty.call(obj, prop)) } }",
+        "class C { #call; foo() { ({}.hasOwnProperty.#call(obj, prop)) } }",
+        "({ foo }.hasOwnProperty.call(obj, prop))",
+        "(Object) => ({}).hasOwnProperty.call(obj, prop)",
+        r#"
+			        let obj = {};
+			        Object.hasOwn(obj,"");
+			        "#,
+        "const hasProperty = Object.hasOwn(object, property);",
+        // "/* global Object: off */
```

Comment:
I have a question.
I don't know how do I handle this comment in oxc ast.
Currently, I commented out.
If I have to handle this test case, please tell me.

Reply:
My approach so far has been "using comments to declare globals is stupid and outdated". Feel free to ignore such cases.